osf/models/spam.py

```diff
@@ -175,114 +158,70 @@ def unspam(self, save=False):
-        if (
-            settings.SPAM_CHECK_ENABLED and
-            'headers' in self.spam_data and self.spam_status in [SpamStatus.UNKNOWN, SpamStatus.HAM] and
-            train_akismet
-        ):
-            client = _get_akismet_client()
-            client.submit_spam(
-                user_ip=self.spam_data['headers']['Remote-Addr'],
-                user_agent=self.spam_data['headers'].get('User-Agent'),
-                referrer=self.spam_data['headers'].get('Referer'),
-                comment_content=self.spam_data['content'],
-                comment_author=self.spam_data['author'],
-                comment_author_email=self.spam_data['author_email'],
+        elif train_spam_services and self.spam_data:
+            submit_spam.apply_async(
+                kwargs=dict(
+                    guid=self.guids.first()._id,
+                )
             )
-            logger.info('confirm_spam update sent')
+
         self.spam_status = SpamStatus.SPAM
         if save:
             self.save()
 
     @abc.abstractmethod
-    def check_spam(self, user, saved_fields, request_headers, save=False):
+    def check_spam(self, user, saved_fields, request, save=False):
         """Must return is_spam"""
         pass
 
-    def do_check_spam(self, author, author_email, content, request_headers, update=True):
+    def do_check_spam(self, author, author_email, content, request_headers):
         if self.is_hammy:
             return False
         if self.is_spammy:
             return True
 
+        request_kwargs = {
+            'remote_addr': request_headers.get('Remote-Addr') or request_headers['Host'],  # for local testing
+            'user_agent': request_headers.get('User-Agent'),
+            'referer': request_headers.get('Referer'),
+        }
+        request_kwargs.update(request_headers)
+
         check_resource_for_domains.apply_async(
             kwargs=dict(
                 guid=self.guids.first()._id,
                 content=content,
             )
         )
-
-        akismet_client = _get_akismet_client()
-        oopspam_client = _get_oopspam_client()
-        remote_addr = request_headers.get('Remote-Addr') or request_headers['Host']  # for local testing
-        user_agent = request_headers.get('User-Agent')
-        referer = request_headers.get('Referer')
-        akismet_is_spam, pro_tip = akismet_client.check_comment(
-            user_ip=remote_addr,
-            user_agent=user_agent,
-            referrer=referer,
-            comment_content=content,
-            comment_author=author,
-            comment_author_email=author_email
-        )
-
-        try:
-            oopspam_is_spam, oopspam_details = oopspam_client.check_content(
-                user_ip=remote_addr,
-                content=content
+        if settings.SPAM_SERVICES_ENABLED:
+            for key, value in request_kwargs.items():
+                request_kwargs[key] = ensure_str(value)
```

Comment:
Sometime the request info is produced as bytes, which causes the Celery task to fail because the signature isn't serializable. It's difficult to predict when the request object is going to have that info so we ensure everything here.

Reply:
ahh, I missed L207. Thanks! There still might be a one-loop way to do this, but it's not urgent